tests/jointcalTestBase.py

```diff
@@ -0,0 +1,311 @@
+            good &= (cat.get('base_ClassificationExtendedness_value') == 0)
+            matches = table.matchRaDec(reference, cat[good], self.matchRadius)
+            for m in matches:
+                deltas[m[0].getId()].append(m[2])
+        return deltas
+
+    def _make_visit_catalogs(self, catalogs, visits):
+        """
+        Return a dict of visit: catalog of all sources from all CCDs of that visit.
+
+        @param catalogs (list of SourceCatalogs) catalogs to combine into per-visit catalogs.
+        @param visits (list) list of visit identifiers, one-to-one correspondant with catalogs.
+        """
+        visit_dict = {v: table.SourceCatalog(catalogs[0].schema) for v in self.visitList}
+        for v, cat in zip(visits, catalogs):
+            visit_dict[v].extend(cat)
+        # We want catalog contiguity to do object selection later.
+        for v in visit_dict:
+            visit_dict[v] = visit_dict[v].copy(deep=True)
+
+        return visit_dict
+
+    def _do_plots(self, dataRefs, visitCatalogs,
+                  old_relative, old_absolute, new_relative, new_absolute,
+                  old_rel_total, old_abs_total, new_rel_total, new_abs_total):
+        """Make plots of various quantites to help with debugging."""
+        import matplotlib.pyplot as plt
+        plt.ion()
+
+        plot_all_wcs_deltas(plt, dataRefs, visitCatalogs, self.old_wcss)
+
+        old_rms_relative = rms_per_star(old_relative)
+        old_rms_absolute = rms_per_star(old_absolute)
+        new_rms_relative = rms_per_star(new_relative)
+        new_rms_absolute = rms_per_star(new_absolute)
+        print(len(dataRefs))
+        print("relative RMS (old, new):", old_rel_total*arcsec_per_radian,
+              new_rel_total*arcsec_per_radian)
+        print("absolute RMS (old, new):", old_abs_total*arcsec_per_radian,
+              new_abs_total*arcsec_per_radian)
+        plot_rms_histogram(plt, old_rms_relative, old_rms_absolute, new_rms_relative, new_rms_absolute,
+                           old_rel_total, old_abs_total, new_rel_total, new_abs_total)
+
+        # So one can muck-about with things after plotting...
+        import ipdb
+        ipdb.set_trace()
+
+
+def rms_per_star(data):
+    """
+    Compute the RMS/catalog star
+
+    @param data (dict) dict of starID:list(measurement deltas for that star)
+
+    @return (np.array) the rms for each starID (no guaranteed order)
+    """
+    result = np.empty(len(data))
+    for i, k in enumerate(data):
+        result[i] = np.sqrt(np.mean(np.array(data[k])**2))
+    return result
+
+
+def rms_total(data):
+    """
+    Compute the total rms across all stars.
+
+    @param data (dict) dict of starID:list(measurement deltas for that star)
+
+    @return (int) the total rms across all stars
+    """
+    n = 0
+    total = 0
+    for i, k in enumerate(data):
+        total += sum(np.array(data[k])**2)
+        n += len(data[k])
```

Comment:
How about:

    total = sum(np.array(dd)**2 for dd in data.itervalues())
    n = sum(len(dd) for dd in data.itervalues())

At the very least, you don't need the `enumerate` and you don't care about the dict key, only the values.